python/pyiceberg/avro/encoder.py

```diff
@@ -64,112 +65,40 @@ def write_decimal_bytes(self, datum: decimal.Decimal) -> None:
-
-        unscaled_datum = 0
-        for digit in digits:
-            unscaled_datum = (unscaled_datum * 10) + digit
-
-        bits_req = unscaled_datum.bit_length() + 1
-        size_in_bits = size * 8
-        offset_bits = size_in_bits - bits_req
-
-        mask = 2**size_in_bits - 1
-        bit = 1
-        for _ in range(bits_req):
-            mask ^= bit
-            bit <<= 1
-
-        if bits_req < 8:
-            bytes_req = 1
-        else:
-            bytes_req = bits_req // 8
-            if bits_req % 8 != 0:
-                bytes_req += 1
-        if sign:
-            unscaled_datum = (1 << bits_req) - unscaled_datum
-            unscaled_datum = mask | unscaled_datum
-            for index in range(size - 1, -1, -1):
-                bits_to_write = unscaled_datum >> (8 * index)
-                self.write(bytearray([bits_to_write & 0xFF]))
-        else:
-            for _ in range(offset_bits // 8):
-                self.write(b"\x00")
-            for index in range(bytes_req - 1, -1, -1):
-                bits_to_write = unscaled_datum >> (8 * index)
-                self.write(bytearray([bits_to_write & 0xFF]))
+        unscaled_datum = decimal_to_unscaled(datum)
+        bytes_datum = unscaled_datum.to_bytes(length=size, byteorder="big", signed=True)
+        self.write(bytes_datum)
 
     def write_bytes(self, b: bytes) -> None:
         """Bytes are encoded as a long followed by that many bytes of data."""
         self.write_int(len(b))
-        self.write(struct.pack(f"{len(b)}s", b))
-
-    def write_bytes_fixed(self, b: bytes) -> None:
-        """Writes fixed number of bytes."""
-        self.write(struct.pack(f"{len(b)}s", b))
+        self.write(b)
 
     def write_utf8(self, s: str) -> None:
         """A string is encoded as a long followed by that many bytes of UTF-8 encoded character data."""
         self.write_bytes(s.encode("utf-8"))
 
-    def write_date_int(self, d: date) -> None:
-        """
-        Encode python date object as int.
-
-        It stores the number of days from the unix epoch, 1 January 1970 (ISO calendar).
-        """
-        self.write_int(date_to_days(d))
-
-    def write_time_millis_int(self, dt: time) -> None:
-        """
-        Encode python time object as int.
+    def write_uuid(self, uuid: UUID) -> None:
+        """Write UUID as a fixed[16].
 
-        It stores the number of milliseconds from midnight, 00:00:00.000
+        The uuid logical type represents a random generated universally unique identifier (UUID).
+        An uuid logical type annotates an Avro string. The string has to conform with RFC-4122.
         """
-        self.write_int(int(time_object_to_micros(dt) / 1000))
+        if len(uuid.bytes) != 16:
+            raise ValueError(f"Expected UUID to have 16 bytes, got: len({uuid.bytes!r})")
+        return self.write(uuid.bytes)
 
-    def write_time_micros_long(self, dt: time) -> None:
+    def write_time_micros(self, dt: time) -> None:
```

Comment:
Do we need a method that accepts a `time` here? By this point, I think we should only have `int` objects because they should be translated into the internal representation (micros from midnight for time, micros from epoch for timestamp, days from epoch for date).

Reply:
I agree, this should be handled by the writer, not the encoder. Another one to clean up, less is more :)